server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -480,6 +503,16 @@ async def _archive_model_endpoint(
                     uids=uid_to_delete,
                     project=model_endpoint.metadata.project,
                 )
+                feature_set_uids = [
+                    f"{framework.db.sqldb.db.unversioned_tagged_object_uid_prefix}{uid}_"
+                    for uid in old_uids
+                ]
+                await run_in_threadpool(
+                    framework.utils.singletons.db.get_db().delete_feature_sets,
+                    session=db_session,
+                    project=model_endpoint.metadata.project,
+                    uids=feature_set_uids,
+                )
```

Comment:
Maybe its better to put it inside `_delete_model_endpoint_monitoring_infra`, and to avoid code duplication